src/rules/noObjectLiteralTypeAssertionRule.ts

```diff
@@ -58,6 +58,10 @@ function walk(ctx: Lint.WalkContext<void>): void {
         if (
             isAssertionExpression(node) &&
             node.type.kind !== ts.SyntaxKind.AnyKeyword &&
+            // Compare with UnknownKeyword if using TS 3.0 or above
+            (!!(ts.SyntaxKind as any).UnknownKeyword ?
+             (node.type.kind !== (ts.SyntaxKind as any).UnknownKeyword) :
+             true) &&
```

Comment:
@NN--- if you change the `true` to `node.type.getText(ctx.sourceFile) !== "unknown"` the `<unknown> {};` line will be properly linted, so you can remove the expected error from it.

Reply:
Just to make sure, in TS 2 there is no unknown keyword and this rule can be abused.
I think these cases are rare.

```ts
type unknown = {};
{} as unknown // Not what we wanted.
```

Reply:
My point is that you could make anything in TS 2.0 as unknown is not a keyword.
Perhaps not a clever idea but still :)

Reply:
Yes understood - and I think that's ok for this rule because, as you noted, it's not a clever idea.